### PR TITLE
fix(theme): render proper 404 page as a content page

### LINKED ISSUE #403: 404 layout broken
The layout of the gatsbyJS 404 page is broken,  in some responsive breakpoints it at least says "page not found" but in the main view it's completely broken. 

Please note:
 * this refers to production builds
 * the gatsbyjs 404 does not appear when directly entering a wrong URL.  But when navigating via back / forward the gatsbyJS client side 404 appears. 
 * ideally we also configure `now.json` to a proper 404.   Let's get it first set up correct in this repo and then do it in the private content repos. 

image

### PR DESCRIPTION
**Pull request by emmenko**
Fixes #403 

image

### CODE REVIEW

File: now.json [hunk 1]
```diff
@@ -13,6 +13,11 @@
     },
     {
       "handle": "filesystem"
+    },
+    {
+      "src": "/([a-z]+)/.*",
+      "status": 404,
+      "dest": "/$1/404"
```
Comment (emmenko) [inline now.json:20]: This should do the trick for the ZEIT routing.

Comment (nkuehn) [reply now.json:20]: In my migration branch I opted to always point to the 404 of the `/docs` homepage path ( `/docs/404` ) - that also catches   paths like  `/12`  without showing the ZEIT 404.  

If we think there's a use case for microsite-specific 404s the approaches can be combined, too - add another route behind yours that points all else to  `/docs/404`

Comment (nkuehn) [reply now.json:20]: PS: obviously, here in the kit repo this is all not a problem.

File: now.json [hunk 2]
```diff
@@ -13,6 +13,11 @@
     },
     {
       "handle": "filesystem"
+    },
+    {
+      "src": "/([a-z]+)/.*",
```
Comment (nkuehn) [inline now.json:18]: this does not catch our website names with dashes. Plus, you need to escape slashes.   

test case: https://commercetools-docs-kit-git-nm-404.commercetools.now.sh/docs-smoke-test/i-do-not-exist

```suggestion
      "src": "\/(\S+?)\/.*",
```
(any non-whitespace character, ungreedily matched until the next forward slash)

https://regex101.com/r/3ntqSz/1

Comment (emmenko) [reply now.json:18]: Thanks 😄

File: now.json (hunk 1 above)
Comment (emmenko) [reply now.json:20]: Yeah, if we have a "home" microsite we can do that as well.

File: now.json [hunk 3]
```diff
@@ -13,6 +13,11 @@
     },
     {
       "handle": "filesystem"
+    },
+    {
+      "src": "\/(\S+?)\/.*",
```
Comment (nkuehn) [inline now.json:18]:
```suggestion
      "src": ""\\\/(\\S+?)\\\/.*"",
```
JSON escaping, no questions...  That's what happens if you try to code in github PR comments :-(

Comment (emmenko) [reply now.json]: Escaping the slash should not be necessary

**Pull request merged**